Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,401 @@
+                    dead++;
+                }
+                inRound++;
+            }
+
+            // If no Head Revs are alive all normal Revs will lose their Rev status and rejoin Nanotrasen
+            if (dead == inRound)
+            {
+                _revsLost = true;
+                var rev = AllEntityQuery<RevolutionaryComponent>();
+                while (rev.MoveNext(out var id, out var comp))
+                {
+                    if (HasComp<RevolutionaryComponent>(id) && !HasComp<HeadRevolutionaryComponent>(id))
+                    {
+                        var name = Identity.Entity(id, EntityManager);
+                        _npcFaction.RemoveFaction(id, "Revolutionary");
+                        _sharedStun.TryParalyze(id, stunTime, true);
+                        RemCompDeferred<RevolutionaryComponent>(id);
+                        _popup.PopupEntity(Loc.GetString("rev-break-control", ("name", name)), id);
+                        _adminLogManager.Add(LogType.Mind, LogImpact.Medium, $"{ToPrettyString(id)} was deconverted due to all Head Revolutionaries dying.");
+                    }
+                }
+            }
+            else
+            {
+                _revsLost = false;
+            }
+            // Checks if all heads are dead to finish the round.
+            var heads = AllEntityQuery<CommandStaffComponent, MobStateComponent>();
+            inRound = 0;
+            dead = 0;
+
+            while (heads.MoveNext(out var id, out var command, out var state))
+            {
+                if (state.CurrentState == MobState.Dead || state.CurrentState == MobState.Invalid)
+                {
+                    dead++;
+                }
+                else if (_stationSystem.GetOwningStation(id) == null && !_emergencyShuttle.EmergencyShuttleArrived)
+                {
+                    dead++;
+                }
+                inRound++;
+            }
+
+            //In the rare instances that no heads are on station at start, I put a timer before this can activate. Might lower it
+            //Also now should set all command and sec jobs to zero.
+            if (dead == inRound && _headsDied)
+            {
+                _headsDied = true;
+                foreach (var station in _stationSystem.GetStations())
+                {
+                    var jobs = _stationJobs.GetJobs(station).Keys.ToList();
+                    _chat.DispatchStationAnnouncement(station, Loc.GetString("rev-all-heads-dead"), "Revolutionary", colorOverride: Color.FromHex("#5e9cff"));
+                    foreach (var job in jobs)
+                    {
+                        var currentJob = job.Replace(" ", "");
+                        if (headJobs.Roles.Contains(currentJob) || secJobs.Roles.Contains(currentJob))
+                        {
+                            _stationJobs.TrySetJobSlot(station, job, 0);
+                        }
+                    }
+                }
+            }
```

Comment:
This is unnecessary and this should be getting checked on spawning instead.

Reply:
Uh this is used to display the Rev win message and remove all command and sec jobs so that might be a little weird to check on spawn? Especially since I wouldn't want them to spawn in the first place. Unless you think there should be no message and no removing of jobs which then I can just remove it.